### PR TITLE
refactor(client): expose some methods to mock all the https calls

### PR DESCRIPTION
**Pull request by Scra3**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/forestadmin-client/src/build-application-services.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  auth: AuthService;
+} {
+  const optionsWithDefaults = {
+    forestServerUrl: 'https://api.forestadmin.com',
+    permissionsCacheDurationInSeconds: 15 * 60,
+    logger: defaultLogger,
+    ...options,
+  };
+
+  const { getEnvironmentPermissions, getUsers, getRenderingPermissions } = forestServerRepository;
```
Comment (arnaud-moncel) [inline packages/forestadmin-client/src/build-application-services.ts:37]: Think about pass directly the service

**Review by arnaud-moncel — COMMENTED** [review]
💪 first review before tests review

File: packages/forestadmin-client/src/types.ts [hunk 2]
```diff
@@ -133,3 +140,9 @@ export interface ContextVariablesInstantiatorInterface {
     userId: string | number;
   }): Promise<ContextVariables>;
 }
+
+export interface ForestServerRepository {
```
[the remark this answers is not in the document]
Comment (Scra3) [reply packages/forestadmin-client/src/types.ts]: I have rename it by ForestAdminServerInterface.

**Review by arnaud-moncel — APPROVED** [review]
👌

**Pull request merged**